src/ansys/geometry/core/shapes/curves/nurbs.py

```diff
@@ -141,6 +142,48 @@ def from_control_points(
 
         return curve
 
+    @classmethod
+    @check_input_types
+    def fit_curve_from_points(
+        cls,
+        points: list[Point3D],
+        degree: int,
+    ) -> "NURBSCurve":
+        """Fit a NURBS curve to a set of points.
+
+        Parameters
+        ----------
+        points : list[Point3D]
+            Points to fit the curve to.
+        degree : int
+            Degree of the curve.
+
+        Returns
+        -------
+        NURBSCurve
+            Fitted NURBS curve.
+
+        """
+        from geomdl import fitting
+
+        # Convert points to a format suitable for the fitting function
+        converted_points = []
+        for pt in points:
+            pt = (pt.x.m, pt.y.m, pt.z.m)
+            converted_points.append(pt)
```

Comment:
```suggestion
            pt_raw = [*pt]
            converted_points.append(pt_raw)
```
Use the raw values under the ndarray... otherwise there will be discrepancies. The raw values are in default units.

See this example:

![image](https://github.com/user-attachments/assets/695e59a8-a351-4045-b08b-4b99713eb9c3)